Sources/Sentry/Profiling/ThreadMetadataCache.mm

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Specto Inc. All rights reserved.
+
+#include "ThreadMetadataCache.h"
+
+#include "ThreadHandle.h"
+#include "StackBounds.h"
+#include "SpectoProtoPolyfills.h"
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+using namespace specto;
+
+namespace {
+// QoS data collection is disabled, see the comments below for the reasoning.
```

Comment:
`l`: My personal preference is to remove commented code and add it from the Git history if we need it again.